pkg/server/router/common.go

```diff
@@ -31,25 +31,40 @@ import (
 // to Router and Gateway.
 type commonRouter struct {
 	payloadFunc      func(req *http.Request) (p *Payload, err error)
-	matchHandlerFunc func(req *http.Request, p *Payload) (h Handler, pp []Processor)
+	matchHandlerFunc func(path string, method string, p *Payload) (h Handler, pp []Processor)
```

Comment:
I am not 100% sure, but I remember Boris asked why do we have both action name in url path and payload, and the answer was like because sometimes we don't have the payload (e.g. a GET request). Create a fake payload from url when payload is nil?